client/history/metricClient.go

```diff
@@ -614,8 +617,8 @@ func (c *metricClient) finishMetricsRecording(
 	err error,
 ) {
 	if err != nil {
-		c.logger.Error("history client encountered error", tag.Error(err))
-		scope.IncCounter(metrics.ClientFailures)
+		c.throttledLogger.Error("history client encountered error", tag.Error(err))
+		scope.Tagged(metrics.ServiceErrorTypeTag(err)).IncCounter(metrics.ClientFailures)
```

Comment:
we should add field for error type to the log line that is the same as the tag on metrics so when we observer some error type from metrics, we can filter on log for that type.